kbchat/kbchat_test.go

```diff
@@ -103,82 +103,290 @@ func deleteWorkingDir(workingDir string) error {
 
-func TestSendMessage(t *testing.T) {}
+func TestSendMessageByConvID(t *testing.T) {
+	alice, _, dir, oneOnOneChannel, _ := testSetup()
+	text := "test SendMessageByConvID"
 
-func TestSendMessageByConvID(t *testing.T) {}
+	// Retrieve conversation ID
+	messages, err := alice.GetTextMessages(oneOnOneChannel, false)
+	require.NoError(t, err)
+	convID := messages[0].ConversationID
+
+	// Send the message
+	res, err := alice.SendMessageByConvID(convID, text)
+	require.NoError(t, err)
+	require.Greater(t, res.Result.MsgID, 0)
+
+	// Read it to confirm it sent
+	messages, err = alice.GetTextMessages(oneOnOneChannel, false)
+	require.NoError(t, err)
+	sentMessage := messages[0]
+	require.Equal(t, sentMessage.Content.Type, "text")
+	require.Equal(t, sentMessage.Content.Text.Body, text)
+	require.Equal(t, sentMessage.MsgID, res.Result.MsgID)
+	testTeardown(alice, dir)
+}
 
 func TestSendMessageByTlfName(t *testing.T) {
-	tlfName := fmt.Sprintf("%s,%s", kbc.Username(), "kb_monbot")
-	res, err := kbc.SendMessageByTlfName(tlfName, "test")
+	alice, _, dir, oneOnOneChannel, _ := testSetup()
+	text := "test SendMessageByTlfName"
+
+	// Send the message
+	res, err := alice.SendMessageByTlfName(oneOnOneChannel.Name, text)
 	require.NoError(t, err)
 	require.Greater(t, res.Result.MsgID, 0)
+
+	// Read it to confirm it sent
+	messages, err := alice.GetTextMessages(oneOnOneChannel, false)
+	require.NoError(t, err)
+	sentMessage := messages[0]
+	require.Equal(t, sentMessage.Content.Type, "text")
+	require.Equal(t, sentMessage.Content.Text.Body, text)
+	require.Equal(t, sentMessage.MsgID, res.Result.MsgID)
+	testTeardown(alice, dir)
 }
 
-func TestSendMessageByTeamName(t *testing.T) {}
+func TestSendMessageByTeamName(t *testing.T) {
+	alice, _, dir, _, teamChannel := testSetup()
+	text := "test SendMessageByTeamName"
 
-func TestSendAttachmentByTeam(t *testing.T) {}
+	// Send the message
+	res, err := alice.SendMessageByTeamName(teamChannel.Name, text, &teamChannel.TopicName)
+	require.NoError(t, err)
+	require.Greater(t, res.Result.MsgID, 0)
 
-func TestReactByChannel(t *testing.T) {}
+	// Read it to confirm it sent
+	messages, err := alice.GetTextMessages(teamChannel, false)
+	require.NoError(t, err)
+	sentMessage := messages[0]
+	require.Equal(t, sentMessage.Content.Type, "text")
+	require.Equal(t, sentMessage.Content.Text.Body, text)
+	require.Equal(t, sentMessage.MsgID, res.Result.MsgID)
+	testTeardown(alice, dir)
+}
+
+func TestSendAttachmentByTeam(t *testing.T) {
+	alice, _, dir, _, teamChannel := testSetup()
+	// Create a test file
+	fileName := "kb-attachment.txt"
+	location := path.Join(os.TempDir(), fileName)
+	data := []byte("My super cool attachment")
+	err := ioutil.WriteFile(location, data, 0644)
+	require.NoError(t, err)
+
+	// Send the message
+	title := "test SendAttachmentByTeam"
+	res, err := alice.SendAttachmentByTeam(teamChannel.Name, location, title, &teamChannel.TopicName)
+	require.NoError(t, err)
+	require.Greater(t, res.Result.MsgID, 0)
```

Comment:
is there anything more you can assert here?

Reply:
not really, without changing or adding to the current bot functions/types. `GetTextMessages` filters out all non-text messages, and we currently don't have any types for attachment messages.

The latter problem would be fixed by generating bot types, so I figured we could revisit this test then.